### PR TITLE
[image_picker] Update NSPhotoLibraryUsageDescription description in README

### PR DESCRIPTION
**Pull request by jmagman**
Fixes https://github.com/flutter/flutter/issues/113603

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [relevant style guides] and ran [the auto-formatter]. (Unlike the flutter/flutter repo, the flutter/plugins repo does use `dart format`.)
- [x] I signed the [CLA].
- [x] The title of the PR starts with the name of the plugin surrounded by square brackets, e.g. `[shared_preferences]`
- [x] I listed at least one issue that this PR fixes in the description above.
- [x] I updated `pubspec.yaml` with an appropriate new version according to the [pub versioning philosophy], or this PR is [exempt from version changes].
- [x] I updated `CHANGELOG.md` to add a description of the change, [following repository CHANGELOG style].
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] I added new tests to check the change I am making, or this PR is [test-exempt].
- [x] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/plugins/blob/main/CONTRIBUTING.md
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[relevant style guides]: https://github.com/flutter/plugins/blob/main/CONTRIBUTING.md#style
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat
[pub versioning philosophy]: https://dart.dev/tools/pub/versioning
[exempt from version changes]: https://github.com/flutter/flutter/wiki/Contributing-to-Plugins-and-Packages#version-and-changelog-updates
[following repository CHANGELOG style]: https://github.com/flutter/flutter/wiki/Contributing-to-Plugins-and-Packages#changelog-style
[the auto-formatter]: https://github.com/flutter/plugins/blob/main/script/tool/README.md#format-code
[test-exempt]: https://github.com/flutter/flutter/wiki/Tree-hygiene#tests

### CODE REVIEW

File: packages/image_picker/image_picker/README.md [hunk 1]
```diff
@@ -23,7 +23,6 @@ As a result of implementing PHPicker it becomes impossible to pick HEIC images o
 Add the following keys to your _Info.plist_ file, located in `<project root>/ios/Runner/Info.plist`:
 
 * `NSPhotoLibraryUsageDescription` - describe why your app needs permission for the photo library. This is called _Privacy - Photo Library Usage Description_ in the visual editor.
-  * This permission is not required for image picking on iOS 11+ if you pass `false` for `requestFullMetadata`.
```
Comment (stuartmorgan) [inline packages/image_picker/image_picker/README.md:26]: LGTM if you want to remove it fully, but maybe instead we should leave it but change to something like "This permission will not be requested if you always pass `false` for `requestFullMetadata`, but App Store policy requires including the plist entry."

That way people (including future us, potentially) won't think it's a mistake that we're listing it here when they read the code documentation that talks about how that flag interacts with permissions.

**Review by stuartmorgan — APPROVED** [review]

**Comment by stuartmorgan** [comment]
Did you want to land this as is, or update it per the comment?

**Comment by jmagman** [comment]
> Did you want to land this as is, or update it per the comment?

I need to update, haven't gotten around to it though.

**Comment by stuartmorgan** [comment]
@jmagman FYI I'm planning on doing the repo merge in the next 1-2 weeks, so if you can find time to update the PRs you have open here it'll save you some PR migration work.

**Comment by jmagman** [comment]
I'm going to merge this before the repo migration happens.  @vashworth we can revisit this wording when you look into the metadata default behavior.

**Pull request merged**